emcc.py

```diff
@@ -1054,6 +1054,11 @@ def check(input_file):
     elif shared.Settings.SIDE_MODULE:
       assert not shared.Settings.MAIN_MODULE
       options.memory_init_file = False # memory init file is not supported with asm.js side modules, must be executable synchronously (for dlopen)
+      # SIDE_MODULE+WASM combo unambiguously specifies what kind of file we want to output, so we don't have to fiddle with file extension
+      if shared.Settings.WASM:
+        wasm_binary_target = target
+      else:
+        js_target = target
```

Comment:
why is this line needed, the js path should still work as before?

Reply:
For consistency, as one might want to always build their side modules to `.so` files, regardless of the build type (wasm or asmjs), to make the filename to `dlopen` predictable. Not really attached to that though.